presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -399,13 +400,16 @@ public synchronized void parentTasksAdded(List<TaskId> parentTasks, boolean noMo
                 newOutputBuffers = newOutputBuffers.withNoMoreBufferIds();
             }
         }
-        else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH) {
+        else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH || fragment.getOutputPartitioning() == OutputPartitioning.HASH_NULL_EVERYWHERE) {
             checkArgument(noMoreParentNodes, "Hash partitioned output requires all parent nodes be added in a single call");
 
+            NullPartitionPolicy nullPartitionPolicy = fragment.getOutputPartitioning() == OutputPartitioning.HASH ? NullPartitionPolicy.ONE_PLACE : NullPartitionPolicy.EVERYWHERE;
```

Comment:
Alternatively, make the null replication policy a separate property of fragment and decouple it from OutputPartitioning.